### ISSUE TITLE
Overfit Goldberg Variations

### ISSUE THREAD

**Issue by purzelrakete** [feldberlin/wavenet#29]
## What

Try to overfit the the Goldberg Variations using the `Tracks` dataset. We can already overfit fine on a single track, the Goldberg Variations Aria, but not on 2017 maestro. Let's try to use the same track, but read using the full datasets pipeline. The single track dataset is using a different code path, since it's held in memory.

Keep in mind that `Track` is heavily overlapping examples. Aria itself is  299.52 seconds long. In `Track`, 80% of this (239.62 seconds) is assigned to the trainset. This is turned into 466 seconds in `Track`, and truncated to 299 seconds in `Tracks`. Testing on `Tracks` vs `Track` should be made comparable in terms of what is shown to the network. 

Need to check, but I think this works out to ~100 * ~240 unique seconds shown to the model in `Track` across 50 epochs.

## Hypothesis

I expect to be able to overfit the Aria as before, using no batch norm. I also expect to be able to overfit a handful of tracks from the same Goldberg Variations recording. 

## Results

Write up the results of your experiment once it has completed and has been analysed. Include links to the treatment run, and also to the baseline if appropriate.

## Acceptance Criteria

- [x] Train on full aria using `Track` codepath
- [x] Train on aria using the `Tracks` codepath
- [x] Train on the first 4 variations

**Comment by purzelrakete** [comment]
## Experiment

```
bin/train fromdir  -p root_dir /srv/datasets/goldberg/aria -p cache_dir /srv/datasets-ssd/goldberg/aria
```
```

**Comment by purzelrakete** [comment]
## Results

Training likelihoods only, since we're only interested in overfitting. 

|                   | training nll     |
|-----------|-----------|
| baseline    | 0.00065      |
| aria | 0.00228         |
| four | 0.00976         |



Audio quality is still a bit worse, which I attribute to the fact that overlapping improves the problems discussed in #7. The beginning of the sample sounds great, but then it trails off a bit.



We can say that

- We managed to reproduce the baseline tracks result with perfect sounding audio generated
- Using batch norm, our loss is 100x higher on the trainset
- Using batch norm, our test loss is much lower

## Insights

These results aren't very surprising. Batch norm has a regularising effect, which explains the better test likelihood, and also the worse training error. 

## Open questions

- How close are we to generating audio without overfitting?
